packages/web-component/src/lib/descope-wc/BaseDescopeWc.ts

```diff
@@ -597,6 +620,15 @@ class BaseDescopeWc extends HTMLElement {
         tenant: this.tenant,
         redirectUrl: this.redirectUrl,
         locale: this.locale,
+        form: Object.entries(this.form || {}).reduce(
```

Comment:
Why do we need to generate the `form.` entries?

In addition, why do we need to add it to the flowState?
We can just merge it with the screen state when relevant, WDYT?

Reply:
1. due to a small gap where custom inputs are named `form.custom` we attempt to apply to custom inputs on any form key. so if you do `form={{ custom: "test" }}` it will always map to `form.custom` within the form.

2. good point, i can always use the `this.form` in init and `onFlowChange`

Reply:
Regarding 1, 
We do not get the custom inputs on the flows api response right now, not sure if this is important or not

When looking on the flow response, I see that all flow inputs are directly under form, for example:
<img width="506" alt="image" src="https://github.com/descope/descope-js/assets/11805663/82f5a072-e9e6-4de8-b448-6473f48d4109">

So after this change, we will have 
```
    "form": {
        "email": "nir@descope.com",
        "form.email": "nir@descope.com"
    }
```

Is this what we want to achieve? 


Regarding 2, 
I think it's better, it will simplify the code a bit

Reply:
2 done 👍 
regarding 1: yes, this is what we want to achieve because the custom inputs are already prefixed with form to avoid overriding the original properties such as email. i didn't want to modify the custom inputs so we will autofill the form for now.
<img width="1218" alt="image" src="https://github.com/descope/descope-js/assets/18641362/5361f715-99f5-437a-a431-0daafeee05ef">